packages/frontend-2/lib/server-management/helpers/types.ts

```diff
@@ -0,0 +1,16 @@
+import { Get } from 'type-fest'
+import { UserListQuery } from '~~/lib/common/generated/gql/graphql'
+
+export type UserItem = NonNullable<Get<UserListQuery, 'admin.userList.items[0]'>>
+
+export interface CTA {
+  type: 'button' | 'link'
+  label: string
+  action: () => void | Promise<void>
```

Comment:
1. `MaybeAsync<void>`
2. I saw the same type duplicated in one or more components, just import this there instead (the Button type as well, IIRC)